Source/SwiftLintBuiltInRules/Rules/Lint/MarkRule.swift

```diff
@@ -9,239 +10,131 @@ struct MarkRule: CorrectableRule {
-            Example("↓//  MARK: comment"): Example("// MARK: comment"),
-            Example("↓//MARK: - comment"): Example("// MARK: - comment"),
-            Example("↓// MARK:- comment"): Example("// MARK: - comment"),
-            Example("↓// MARK: -comment"): Example("// MARK: - comment"),
-            Example("↓// MARK: -  comment"): Example("// MARK: - comment"),
-            Example("↓// Mark: comment"): Example("// MARK: comment"),
-            Example("↓// Mark: - comment"): Example("// MARK: - comment"),
-            Example("↓// MARK - comment"): Example("// MARK: - comment"),
-            Example("↓// MARK : comment"): Example("// MARK: comment"),
-            Example("↓// MARKL:"): Example("// MARK:"),
-            Example("↓// MARKL: -"): Example("// MARK: -"),
-            Example("↓// MARKK "): Example("// MARK: "),
-            Example("↓// MARKK -"): Example("// MARK: -"),
-            Example("↓/// MARK:"): Example("// MARK:"),
-            Example("↓/// MARK comment"): Example("// MARK: comment"),
-            issue1029Example: issue1029Correction,
-            issue1749Example: issue1749Correction
-        ]
+        nonTriggeringExamples: MarkRuleExamples.nonTriggeringExamples,
+        triggeringExamples: MarkRuleExamples.triggeringExamples,
+        corrections: MarkRuleExamples.corrections
     )
+}
 
-    private let spaceStartPattern = "(?:\(nonSpaceOrTwoOrMoreSpace)\(mark))"
-
-    private let endNonSpacePattern = "(?:\(mark)\(nonSpace))"
-    private let endTwoOrMoreSpacePattern = "(?:\(mark)\(twoOrMoreSpace))"
-
-    private let invalidEndSpacesPattern = "(?:\(mark)\(nonSpaceOrTwoOrMoreSpace))"
-
-    private let twoOrMoreSpacesAfterHyphenPattern = "(?:\(mark) -\(twoOrMoreSpace))"
-    private let nonSpaceOrNewlineAfterHyphenPattern = "(?:\(mark) -[^ \n])"
-
-    private let invalidSpacesAfterHyphenPattern = "(?:\(mark) -\(nonSpaceOrTwoOrMoreSpaceOrNewline))"
-
-    private let invalidLowercasePattern = "(?:// ?[Mm]ark:)"
-
-    private let missingColonPattern = "(?:// ?MARK[^:])"
-    // The below patterns more specifically describe some of the above pattern's failure cases for correction.
-    private let oneOrMoreSpacesBeforeColonPattern = "(?:// ?MARK +:)"
-    private let nonWhitespaceBeforeColonPattern = "(?:// ?MARK\\S+:)"
-    private let nonWhitespaceNorColonBeforeSpacesPattern = "(?:// ?MARK[^\\s:]* +)"
-    private let threeSlashesInsteadOfTwo = "/// MARK:?"
-
-    private var pattern: String {
-        return [
-            spaceStartPattern,
-            invalidEndSpacesPattern,
-            invalidSpacesAfterHyphenPattern,
-            invalidLowercasePattern,
-            missingColonPattern,
-            threeSlashesInsteadOfTwo
-        ].joined(separator: "|")
+private extension MarkRule {
+    final class Visitor: ViolationsSyntaxVisitor<ConfigurationType> {
+        override func visit(_ token: TokenSyntax) -> SyntaxVisitorContinueKind {
```

Comment:
You can use `visitPost` as well. It allows you to omit the `return` and the semantic should be the same as tokens don't have children.